internal/service/project/resource_project.go

```diff
@@ -269,7 +269,7 @@ func (r *projectRS) Create(ctx context.Context, req resource.CreateRequest, resp
 
 		_, _, err := connV2.TeamsApi.AddAllTeamsToProject(ctx, project.GetId(), NewTeamRoleList(ctx, teams)).Execute()
 		if err != nil {
-			errd := deleteProject(ctx, r.Client.AtlasV2, project.Id)
+			errd := deleteProject(ctx, r.Client.AtlasV2, project.GetId())
```

Comment:
can we have a test that fails before this change and passes after?

Reply:
we now have. The call we mock in the test is only mocked with correct projectId, so in case the string is malformed, tests fail. See image
![Screenshot 2023-12-11 at 11 14 26](https://github.com/mongodb/terraform-provider-mongodbatlas/assets/55513886/89614330-d5f7-4fe2-a4f7-928497c272cb)